### ISSUE TITLE
Cilium health check fails towards control plane node

### ISSUE THREAD

**Issue by maximumG** [kubernetes/kops#15329]
/kind bug

**1. What `kops` version are you running?**

`Client version: 1.26.2 (git-v1.26.2)`

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

`v1.25.9`

**3. What cloud provider are you using?**

AWS

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

Trying to follow [kOps Getting Started](https://kops.sigs.k8s.io/getting_started/aws/) with cilium enabled

```
kops create cluster --name=${NAME} --cloud=aws --zones=us-west-1a --discovery-store=s3://xxx-xxx-xxx/${NAME}/discovery
```

**5. What happened after the commands executed?**

cilium-agent running on worker nodes reports that the health check towards control-plane nodes fails.

```
$ cilium status --verbose
  Name                              IP                Node          Endpoints
  i-018a7b5c13a2332ae               10.4.4.191        unreachable   reachable
  i-08674cf4137ca05a5               10.4.5.111        unreachable   reachable
  i-0a2ccc6a8bc4dc881               10.4.6.163        unreachable   reachable
  
  
$ cilium-health status
  [output omitted]
  i-08674cf4137ca05a5:
    Host connectivity to 10.4.5.111:
      ICMP to stack:   Connection timed out
      HTTP to agent:   OK, RTT=298.557us
    Endpoint connectivity to 100.96.47.26:
      ICMP to stack:   OK, RTT=1.46466ms
      HTTP to agent:   OK, RTT=344.044us
```

However on cilium-agent running on control plane nodes, cilium health checks are working not matter which nodes the target is. **Basically, the AWS security group on control plane nodes does not allow inbound ICMP request from worker nodes.**

![image](https://user-images.githubusercontent.com/4469833/232496289-1aea1595-2833-4e88-a123-3c291e8256ff.png)

**6. What did you expect to happen?**

Kops should have configured the control plane Security group to allow inbound ICMP request from worker nodes.

**7. Please provide your cluster manifest. Execute

 N/A

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**
  
  N/A

**9. Anything else do we need to know?**

N/A

**Comment by olemarkus** [comment]
Do you actually need these health checks to pass? Doesn't the HTTP checks supersede the ICMP checks?

**Comment by maximumG** [comment]
> Doesn't the HTTP checks supersede the ICMP checks?

Based on our testing, cilium reports the node as _unreachable_ if one of the 2 health check fails. In our case, because ICMP health check fails the node is considered as unreachable no matter if the HTTP check succeeds.

As we are building monitoring and included the cilium metric `unreachable_nodes` (:link: [here](https://docs.cilium.io/en/v1.13/observability/metrics/#cluster-health)), we always have some false positive unreachable nodes in our graph when dealing with kops cluster.

Basically our baseline for a healthy clusters means that every worker node has 3 unreachable nodes.

![image](https://user-images.githubusercontent.com/4469833/232731857-05fe47ea-3ed4-499c-a95a-20bc2eed5bba.png)
